checkov/terraform/tf_parser.py

```diff
@@ -727,3 +737,24 @@ def load_or_die_quietly(
         logging.debug(f'failed while parsing file {file_path}', exc_info=True)
         parsing_errors[file_path] = e
         return None
+
+
+def __parse_with_timeout(f: TextIO) -> dict[str, list[dict[str, Any]]]:
+    # setting up timeout class
+    timeout_class: Optional[Type[BaseTimeout]] = None
+    if platform.system() == 'Windows':
+        timeout_class = ThreadingTimeout
+    elif threading.current_thread() is threading.main_thread():
+        timeout_class = SignalTimeout
+
+    # if we're not running on the main thread, don't use timeout
+    if not timeout_class:
+        return hcl2.load(f)
+
+    parsing_timeout = int(os.getenv("HCL_PARSE_TIMEOUT_SEC", "10"))
```

Comment:
all env vars in checkov are defined under`checkov/common/util/env_vars_config.py`, please add it there as well and then import it here